### PR TITLE
vimeo-linker: Link vimeo videos to XML via fuzzy matching.

### PR DESCRIPTION
**Pull request by akoehn**
Also links some previously unlinked videos to papers in the anthology.  These are completely unchecked!

The python script has sub-par quality but I wanted to publish it instead of having it lying around.  Especially the handling of XML modifications should definitely be handled through the anthology code.

### CODE REVIEW

File: bin/vimeo-linker.py [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+
+            if res != None:
+                (idx, p) = res
+                if checkVideo(p):
+                    print("video already exists, skipping ...")
+                    continue
+                title = p.get_title("plain")
+                print("found title " + title + " for video " + name)
+                result += title +"\t"+ idx +"\t"+ url +"\t"+ name +"\t"+ p.get_booktitle("plain") + "\n"
+                id_video.append((idx, url))
```
Comment (mjpost) [inline bin/vimeo-linker.py]: An f-string here (since we require Python 3.7 anyway) would make this more readable.

File: bin/vimeo-linker.py [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+venues = set([x[0].split("-")[0] for x in id_video])
+
+id_video_dict = {id: video for id, video in id_video}
+
+paperidre = re.compile(r'.*<url>(\w\d\d-\d\d\d\d)</url>')
+
+has_video = False
+idx = "notset"
+for v in venues:
+    with open("../data/xml/"+v+".xml") as f:
```
Comment (mjpost) [inline bin/vimeo-linker.py]: You’re redefining v which could cause a problem down the road.

File: data/xml/N19.xml [hunk 3]
```diff
@@ -4599,6 +4609,7 @@
       <url>N19-3002</url>
       <attachment type="presentation">N19-3002.Presentation.pdf</attachment>
       <attachment type="note">N19-3002.Note.pdf</attachment>
+      <video href="https://vimeo.com/347400639" tag="video"/>
     </paper>
     <paper id="3">
```
Comment (mjpost) [inline data/xml/N19.xml]: This is a bad match. Maybe the script should print out fuzzy matches so a human can at least skim through them?

File: data/xml/Q18.xml [hunk 4]
```diff
@@ -493,6 +496,7 @@
       <abstract>Can advances in NLP help advance cognitive modeling? We examine the role of artificial neural networks, the current state of the art in many common NLP tasks, by returning to a classic case study. In 1986, Rumelhart and McClelland famously introduced a neural architecture that learned to transduce English verb stems to their past tense forms. Shortly thereafter in 1988, Pinker and Prince presented a comprehensive rebuttal of many of Rumelhart and McClelland’s claims. Much of the force of their attack centered on the empirical inadequacy of the Rumelhart and McClelland model. Today, however, that model is severely outmoded. We show that the Encoder-Decoder network architectures used in modern NLP systems obviate most of Pinker and Prince’s criticisms without requiring any simplification of the past tense mapping problem. We suggest that the empirical performance of modern networks warrants a reexamination of their utility in linguistic and cognitive modeling.</abstract>
       <pages>651–665</pages>
       <url>Q18-1045</url>
+      <video href="https://vimeo.com/305684335" tag="video"/>
     </paper>
     <paper id="46">
```
Comment (mjpost) [inline data/xml/Q18.xml]: This one’s good but I want to comment that I hate how many of the videos have title-final periods.

File: data/xml/N19.xml (hunk 3 above)
Comment (akoehn) [reply data/xml/N19.xml]: Looks like a correct match to me.  All matches are either substring or levenshtein matches, so all of them have to be checked.  I could special-case the ones were substring match means perfect match and exclude them from the list the script generates.

**Comment by akoehn** [comment]
I forgot that the script generates a CSV file which contains the paper title, ACL ID, Vimeo video title and vimeo link.  Given that the script changes the XML correctly, it is probably easier to check the changes with this file:
[matched-videos-csv.txt](https://github.com/acl-org/acl-anthology/files/3396907/matched-videos-csv.txt)

File: data/xml/N19.xml (hunk 3 above)
Comment (mjpost) [reply data/xml/N19.xml]: You’re right. Doing the review in an iPad...

**Comment by mjpost** [comment]
I’ll leave the merge to you.

**Comment by akoehn** [comment]
Found a mismatched Video, the others seem to be ok. Will merge once the checks are ready.

**Comment by mjpost** [comment]
Building and pushing now.

**Pull request merged**